pkg/endpoint/policy.go

```diff
@@ -879,6 +884,71 @@ func (e *Endpoint) runIdentityToK8sPodSync() {
 	)
 }
 
+// This synchronizes the key-value store with a mapping of the endpoint's IP
+// with the numerical ID representing its security identity.
+func (e *Endpoint) runIPIdentitySync(endpointIP addressing.CiliumIP) {
+
+	if endpointIP == nil {
+		return
+	}
+
+	var addressFamily string
+	if endpointIP.IsIPv6() {
+		addressFamily = "ipv6"
+	} else {
+		addressFamily = "ipv4"
+	}
+
+	e.controllers.UpdateController(fmt.Sprintf("sync-%s-identity-mapping (%d)", addressFamily, e.ID),
+		controller.ControllerParams{
+			DoFunc: func() error {
+
+				ipKey := path.Join(ipcache.IPIdentitiesPath, ipcache.AddressSpace, endpointIP.String())
+				e.Mutex.RLock()
+
+				if e.state == StateDisconnected || e.state == StateDisconnecting {
+					log.WithFields(logrus.Fields{logfields.EndpointState: e.state}).
+						Debugf("not synchronizing endpoint IP with kvstore due to endpoint state")
+					return nil
+				}
+
+				if e.SecurityIdentity == nil {
+					e.Mutex.RUnlock()
+					return nil
+				}
+				//identityValue := e.SecurityIdentity.ID.StringID()
+				ipIdPair := identityPkg.IPIdentityPair{
+					IP: endpointIP.IP(),
+					ID: e.SecurityIdentity.ID,
+				}
+
+				// Release lock as we do not want to have long-lasting key-value
+				// store operations resulting in lock being held for a long time.
+				e.Mutex.RUnlock()
+
+				marshaledIpIdPair, err := json.Marshal(ipIdPair)
+				if err != nil {
+					log.Errorf("err: %s", err)
```

Comment:
I don't think you need to log this error. The controller will do it for you.

Reply:
Right you are! Will fix in next upload.

Reply:
Fixed.